#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

# Define package name
PACKAGE = "ros_examples"

# Generae parameter structure
gen = ParameterGenerator()

# Add the following parameters with types, default value, description and min/max ranges
gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)
gen.add("param",   str_t,   0, "string message to publish",  "YoYo")

# Generate the code.
# The third argument is the prefix of the header file to be included in the server node.
exit(gen.generate(PACKAGE, "ros_examples", "Example"))
